### PR TITLE
promisified exec and execFile should return a promise with ChildProcess instance attached

### PR DESCRIPTION
**Pull request by onip**
as per [documentation](https://nodejs.org/dist/latest-v12.x/docs/api/child_process.html#child_process_child_process_exec_command_options_callback) `require('child_process').exec` and `require('child_process').execFile` should return a promise with attached a the `require('child_process').ChildProcess` instance attached to the `.child` property.

this PR is an adaption of the same code used inside nodejs

```
> require('child_process').exec[require('util').promisify.custom].toString ()
'(...args) => {\n' +
  '    let resolve;\n' +
  '    let reject;\n' +
  '    const promise = new Promise((res, rej) => {\n' +
  '      resolve = res;\n' +
  '      reject = rej;\n' +
  '    });\n' +
  '\n' +
  '    promise.child = orig(...args, (err, stdout, stderr) => {\n' +
  '      if (err !== null) {\n' +
  '        err.stdout = stdout;\n' +
  '        err.stderr = stderr;\n' +
  '        reject(err);\n' +
  '      } else {\n' +
  '        resolve({ stdout, stderr });\n' +
  '      }\n' +
  '    });\n' +
  '\n' +
  '    return promise;\n' +
  '  }'
```

### CODE REVIEW

**Comment by onip** [comment]
resolves #881

**Comment by robertsLando** [comment]
@onip Could you fix conflicts please? This is ok to merge once conflicts are fixed and tests pass @leerob

**Comment by onip** [comment]
`master` merged in

File: prelude/bootstrap.js [hunk 1]
```diff
@@ -1660,22 +1660,29 @@ function payloadFileSync(pointer) {
...
-              error.stdout = stdout;
-              error.stderr = stderr;
-              reject(error);
-            } else {
-              resolve({ stdout, stderr });
-            }
-          })
-        );
+    var customPromiseExecFunction = (o) => (...args) => {
+      let resolve, reject;
```
Comment (robertsLando) [inline prelude/bootstrap.js:1664]:
```suggestion
      let resolve;
      let reject;
```

**Review by robertsLando — CHANGES_REQUESTED** [review]

**Review by robertsLando — APPROVED** [review]
LGTM. @leerob merge this when possible please

**Pull request merged**